assets/js/blocks/product-results-count/style.scss

```diff
@@ -5,5 +5,6 @@
 	.woocommerce-result-count {
 		float: none;
 		font-size: inherit;
+		margin: 0;
```

Comment:
For me, `.woocommerce .woocommerce-ordering` also has some bottom margin, should we remove it from there too so both blocks are aligned?

![imatge](https://user-images.githubusercontent.com/3616980/214588682-293adf6c-fb83-4199-a5ef-5c871e0e037d.png)